opencog/guile/SchemeSmobAS.cc

```diff
@@ -277,7 +277,7 @@ SCM SchemeSmob::ss_as(SCM satom)
 /**
  * Return current atomspace for this dynamic state.
  */
-SCM SchemeSmob::atomspace_fluid;
+std::atomic<SCM> SchemeSmob::atomspace_fluid;
 
```

Comment:
This is also a non-sense fix!  Its a fluid, which is another name for TLS or thread-local-storage. It does not make sense to make TLS storage atomic.

Again, the key point is that this storage location is initialized once and only once, and is never written to again.   Because it is a fluid, every thread gets it's own unique copy, making it effectively atomic in each thread.  

Anyway, as far as I am aware, there is only one atomspace in use in the entire system, so all of the fluids have exactly the same atomspace, which is the same atomspace as the cogserver, so even the TLS locations always contain exactly the same data.  So, again, this fix doesn't make any sense.